spec.html

```diff
@@ -12308,50 +12241,35 @@ <h1>
-            1. Else,
-              1. If _O_ is not *undefined*, convert the property named _P_ of object _O_ from an accessor property to a data property. Preserve the existing values of the converted property's [[Configurable]] and [[Enumerable]] attributes and set the rest of the property's attributes to their <emu-xref href="#table-default-attribute-values">default values</emu-xref>.
-          1. Else if IsDataDescriptor(_current_) and IsDataDescriptor(_Desc_) are both *true*, then
-            1. If _current_.[[Configurable]] is *false* and _current_.[[Writable]] is *false*, then
-              1. If _Desc_.[[Writable]] is present and _Desc_.[[Writable]] is *true*, return *false*.
-              1. If _Desc_.[[Value]] is present and SameValue(_Desc_.[[Value]], _current_.[[Value]]) is *false*, return *false*.
-              1. Return *true*.
-          1. Else,
-            1. Assert: ! IsAccessorDescriptor(_current_) and ! IsAccessorDescriptor(_Desc_) are both *true*.
-            1. If _current_.[[Configurable]] is *false*, then
-              1. If _Desc_.[[Set]] is present and SameValue(_Desc_.[[Set]], _current_.[[Set]]) is *false*, return *false*.
-              1. If _Desc_.[[Get]] is present and SameValue(_Desc_.[[Get]], _current_.[[Get]]) is *false*, return *false*.
-              1. Return *true*.
+            1. Let _unsettableFields_ be &laquo; [[Configurable]], [[Enumerable]], [[Get]], [[Set]] &raquo;.
+            1. If ! IsAccessorDescriptor(_current_) is *true*, append [[Value]] and [[Writable]] to _unsettableFields_.
+            1. If _current_.[[Writable]] is present and is *false*, append [[Value]] and [[Writable]] to _unsettableFields_.
+            1. For each field name _field_ of _unsettableFields_, do
+              1. If _Desc_.[[&lt;_field_&gt;]] is present and either _current_.[[&lt;_field_&gt;]] is not present or ! SameValue(_Desc_.[[&lt;_field_&gt;]], _current_.[[&lt;_field_&gt;]]) is *false*, return *false*.
           1. If _O_ is not *undefined*, then
+            1. If ! IsDataDescriptor(_current_) is *true* and ! IsAccessorDescriptor(_Desc_) is *true*, then
+              1. Replace the property named _P_ of object _O_ with an accessor property having [[Configurable]] and [[Enumerable]] attributes as described by _current_ and each other attribute set to its <emu-xref href="#table-object-property-attributes">default value</emu-xref>.
+            1. Else if ! IsAccessorDescriptor(_current_) is *true* and ! IsDataDescriptor(_Desc_) is *true*, then
+              1. Replace the property named _P_ of object _O_ with a data property having [[Configurable]] and [[Enumerable]] attributes as described by _current_ and each other attribute set to its <emu-xref href="#table-object-property-attributes">default value</emu-xref>.
             1. For each field of _Desc_ that is present, set the corresponding attribute of the property named _P_ of object _O_ to the value of the field.
```

Comment:
could these perhaps be rewritten to use `! DefineOwnPropertyOrThrow`, since we know O is configurable?

Reply:
I don't think so, because we're logically one or two layers down from that, setting field-like attributes on the Record-like data structure that represents "a property of an object"... note that ValidateAndApplyPropertyDescriptor is invoked _by_ (and is basically the implementation of) OrdinaryDefineOwnProperty.

Reply:
fair point, thanks